locust_plugins/jmeter_listener.py

```diff
@@ -0,0 +1,250 @@
+"""
+a listener to provide results from locust tests in a JMeter compatible format
+and thereby smooth the transition for JMeter users
+"""
+
+from datetime import datetime
+from time import time
+from locust import events
+import logging
+
+
+class JmeterListener:
+    """
+    create an intance of the listener at the start of a test
+    to create a JMeter style results file
+    different formats can be chosen in initialisation
+    (field_delimiter row_delimiter and timestamp_format)
+    and the number of results to send to a log file at a time (flush_size)
+    by default, it will automatically log results (auto_log=True)
+    if you want to manually override the pass/fail outcome and write a custom-made result
+    set auto_log to false in initialisation and for each task and call add_result
+    with the values you want to record
+    Manual logging has to be used FastHttpUser
+    """
+
+    # holds results until processed
+    csv_results = []
+
+    def __init__(
+            self,
+            field_delimiter=",",
+            row_delimiter="\n",
+            timestamp_format="%Y-%m-%d %H:%M:%S",
+            flush_size=100,
+            auto_log=True,
+    ):
+        # determine whether to auto log requests or do it manually
+        self.auto_log = auto_log
+        # default JMeter field and row delimiters
+        self.field_delimiter = field_delimiter
+        self.row_delimiter = row_delimiter
+        # a timestamp format, others could be added...
+        self.timestamp_format = timestamp_format
+        # how many records should be held before flushing to disk
+        self.flush_size = flush_size
+        # results filename format
+        self.results_timestamp_format = "%Y_%m_%d_%H_%M_%S"
+        self.results_filename = (
+            "results_" + datetime.fromtimestamp(time()).strftime(self.results_timestamp_format) + ".csv"
+        )
+
+        # fields set by default in jmeter
+        self.csv_headers = [
+            "timeStamp",
+            "elapsed",
+            "label",
+            "responseCode",
+            "responseMessage",
+            "threadName",
+            "dataType",
+            "success",
+            "failureMessage",
+            "bytes",
+            "sentBytes",
+            "grpThreads",
+            "allThreads",
+            "Latency",
+            "IdleTime",
+            "Connect",
+        ]
+        self.results_file = self._create_results_log()
+        self.user_count = 0
+        self.user_name = ""
+        events.quitting.add_listener(self._write_final_log)
+        events.init.add_listener(self.on_locust_init)
+
+    def _set_user_name(self, name):
```

Comment:
this method seems unnecessary & could just as well be removed?

Reply:
well, I suppose I'm not using a getter...
I'll remove it